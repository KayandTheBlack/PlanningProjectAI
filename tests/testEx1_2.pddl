(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ cityK cityL - city
    hotelA_A hotelA_B hotelB_A hotelB_B hotelC_A hotelC_B hotelD_A hotelD_B hotelE_A hotelE_B hotelF_A hotelF_B hotelG_A hotelG_B hotelH_A hotelH_B hotelI_A hotelI_B hotelJ_A hotelJ_B hotelK_A hotelK_B hotelL_A hotelL_B - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelK_A cityK)
    (hotelAt hotelK_B cityK)
    (hotelAt hotelL_A cityL)
    (hotelAt hotelL_B cityL)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight Origin cityK)
    (flight Origin cityL)
    (flight cityA cityB)
    (flight cityA cityD)
    (flight cityA cityI)
    (flight cityA cityL)
    (flight cityB cityE)
    (flight cityB cityG)
    (flight cityB cityJ)
    (flight cityB cityL)
    (flight cityC cityA)
    (flight cityC cityB)
    (flight cityC cityJ)
    (flight cityC cityL)
    (flight cityD cityA)
    (flight cityD cityH)
    (flight cityD cityI)
    (flight cityD cityL)
    (flight cityE cityC)
    (flight cityE cityD)
    (flight cityE cityG)
    (flight cityE cityH)
    (flight cityF cityC)
    (flight cityF cityG)
    (flight cityF cityH)
    (flight cityF cityL)
    (flight cityG cityE)
    (flight cityG cityF)
    (flight cityG cityH)
    (flight cityG cityL)
    (flight cityH cityE)
    (flight cityH cityF)
    (flight cityH cityG)
    (flight cityH cityL)
    (flight cityI cityA)
    (flight cityI cityB)
    (flight cityI cityC)
    (flight cityI cityD)
    (flight cityJ cityC)
    (flight cityJ cityF)
    (flight cityJ cityG)
    (flight cityJ cityI)
    (flight cityK cityA)
    (flight cityK cityF)
    (flight cityK cityJ)
    (flight cityK cityL)
    (flight cityL cityA)
    (flight cityL cityB)
    (flight cityL cityC)
    (flight cityL cityI)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 1)
    (= (maxDaysInCity) 2)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (stayDuration cityJ) 0)
    (= (stayDuration cityK) 0)
    (= (stayDuration cityL) 0)

  )

  (:goal
    (and
      (>= (citiesVisited) 6)
      (>= (totalDays) 20)
    )
  )


)
