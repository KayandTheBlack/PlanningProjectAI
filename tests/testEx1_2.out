
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


no metric specified. plan length assumed.

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is  plan length

advancing to distance:   20
                         19
                         18
                         17
                         16
                         15
                         14
                         13
                         12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYL HOTELL_A
        1: INCREASESTAY CITYL
        2: FLYANDSTAY CITYL CITYI HOTELI_A
        3: INCREASESTAY CITYI
        4: FLYANDSTAY CITYI CITYD HOTELD_A
        5: INCREASESTAY CITYD
        6: FLYANDSTAY CITYD CITYH HOTELH_A
        7: INCREASESTAY CITYH
        8: FLYANDSTAY CITYH CITYG HOTELG_A
        9: INCREASESTAY CITYG
       10: FLYANDSTAY CITYG CITYF HOTELF_A
       11: INCREASESTAY CITYF
       12: FLYANDSTAY CITYF CITYC HOTELC_A
       13: INCREASESTAY CITYC
       14: FLYANDSTAY CITYC CITYA HOTELA_A
       15: INCREASESTAY CITYA
       16: FLYANDSTAY CITYA CITYB HOTELB_A
       17: INCREASESTAY CITYB
       18: FLYANDSTAY CITYB CITYJ HOTELJ_A
       19: INCREASESTAY CITYJ
     

time spent:    0.00 seconds instantiating 133 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 38 facts and 132 actions
               0.00 seconds creating final representation with 37 relevant facts, 27 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.00 seconds searching, evaluating 79 states, to a max depth of 0
               0.00 seconds total time

